### PR TITLE
Provide a simple way to test that a field exists in a model

### PR DESCRIPTION
**Pull request by firemankurt**
When creating a behavior class where most models might have for example, a user_id field, it is very helpful to find out if the current $model you are working with has that field so you can fill it or process it in some cases.

For example:
CompanyBehavior.php
             
    protected function createModel() {
		$this->model = new $this->modelClass;
		// set company of new model
		$this->model->company_id = $this->user()->company_id;

		if ($this->hasAttribute('user_id')) {
			$this->model->user_id = $this->user()->id;
		}
		return $this->model;
	}  

This is a resubmital of: https://github.com/octobercms/library/pull/321

### CODE REVIEW

File: src/Database/Model.php [hunk 1]
```diff
@@ -963,4 +963,15 @@ public function hasSetMutator($key)
...
     }
+    
+    
+    /**
+     * Determine if an attribute exists in the model.
+     * note: nullable attributes must set raw value of null before this works
+     * @param  string  $key
+     * @return bool
+     */
+	public function hasAttribute($attr) {
```
Comment (LukeTowers) [inline src/Database/Model.php]: four spaces not tabs

File: src/Database/Model.php [hunk 2]
```diff
@@ -963,4 +963,15 @@ public function hasSetMutator($key)
     {
         return $this->methodExists('set'.Str::studly($key).'Attribute');
     }
+    
+    
+    /**
+     * Determine if an attribute exists in the model.
+     * note: nullable attributes must set raw value of null before this works
+     * @param  string  $key
```
Comment (LukeTowers) [inline src/Database/Model.php]: variable name here is $attr, not $key

Comment (LukeTowers) [reply src/Database/Model.php]: you could have just changed the docblock instead ;)

File: src/Database/Model.php [hunk 3]
```diff
@@ -963,4 +963,15 @@ public function hasSetMutator($key)
...
+     * @return bool
+     */
+    public function hasAttribute($key) {
```
Comment (tschallacka) [inline src/Database/Model.php]: This is not PSR-4 compliant, the `{` should be on the next line.

**Comment by tschallacka** [comment]
It seems that this has been denied for issues by the laravel repo several times   
https://github.com/laravel/framework/pull/22249  
https://github.com/laravel/framework/pull/22250  
https://github.com/laravel/framework/pull/23375  

You should also use `array_key_exists($key, $model->attributesToArray())` as per https://github.com/laravel/framework/pull/22250#issuecomment-349975338 to prevent skipping over the attributes that are retrieveable via getter.

**Comment by LukeTowers** [comment]
@firemankurt if you could make the changes suggested by @tschallacka that would be appreciated. I don't mind being better than Laravel by adding this into October :) 
Will require documentation though.

**Comment by LukeTowers** [comment]
@firemankurt could you add this to the documentation in the Models section?

**Comment by LukeTowers** [comment]
@firemankurt are you still interested in this?

**Comment by firemankurt** [comment]
Yes

**Comment by LukeTowers** [comment]
@firemankurt could you add this to the documentation in the Models section?

**Comment by firemankurt** [comment]
Hey Luke,
I'm still hoping this will be added to October.
Thanks

**Comment by bennothommo** [comment]
@firemankurt we're still waiting on your documentation update for the Models docs. Did you happen to have a look at that?

**Pull request closed without merging**